### PR TITLE
VACMS-0000: Prevent an exception in certain scenarios in magic anchor link JS.

### PR DESCRIPTION
**Pull request by ndouglas**
## Description
**TL;DR**: Check if jQuery finds a matching element before we try to get its offset.

## QA steps

What needs to be checked to prove this works?  
What needs to be checked to prove it didn't break any related things?  
What variations of circumstances (users, actions, values) need to be checked?  

As user _uid_ with _user_role_
1. Do this
   - [ ] Validate that
2. Then
   - [ ] Validate that
3. Then validate Acceptance Criteria from issue
   - [ ] a
   - [ ] b
   - [ ] c

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [ ] `Platform CMS Team`
- [ ] `Sitewide program`
- [ ] `⭐️ Sitewide CMS`
- [ ] `⭐️ Public websites`
- [ ] `⭐️ Facilities`
- [ ] `⭐️ User support`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
- [ ] Yes, and it's written in issue ____ and queued for publication.
  - [ ] Merge and ping @ rachel-kauff so she's ready to publish after deployment
- [ ] Yes, but it hasn't yet been written
  - [ ] Don't merge yet -- ping @ rachel-kauff to prompt her to write and queue content
- [ ] No announcement is needed for this code change.
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

File: docroot/themes/custom/vagovadmin/assets/js/anchor_links.es6.js [hunk 1]
```diff
@@ -25,19 +25,21 @@
    */
   Drupal.behaviors.vagovadminAnchorLinks = {
     attach() {
-      $('a[href^="#"]:not([href^="#edit-group"])').click((event) => {
+      $('a[href^="#"][href!="#"]:not([href^="#edit-group"])').click((event) => {
```
Comment (ndouglas) [inline docroot/themes/custom/vagovadmin/assets/js/anchor_links.es6.js:28]: Ignore the `<a href="#" onclick="javascript:mineMonero();"></a>` type tags.

File: docroot/themes/custom/vagovadmin/assets/js/anchor_links.es6.js [hunk 2]
```diff
@@ -25,19 +25,21 @@
...
         event.preventDefault();
 
         const target = $(event.target).attr("href");
-        const targetOffset = $(target).offset();
-        if (targetOffset) {
-          const scrollToPosition =
-            targetOffset.top - (Drupal.getAdminToolbarHeight() + 10);
+        if ($(target).length) {
```
Comment (ndouglas) [inline docroot/themes/custom/vagovadmin/assets/js/anchor_links.es6.js:32]: check da lengf

**Comment by swirtSJW** [comment]
The code looks good.  But what should I be checking to verify that it works as it did before and fixes the problem.

**Comment by ndouglas** [comment]
@swirtSJW I'll add a behavioral test to confirm that the scrolling behavior works as intended.

In terms of the actual bug this fixes:
1. Go to `/node/add/event`
2. Click 'Add media'
3. Click 'Select file'
4. Observe that an uncaught JS error is reported in the console.

The above doesn't really break anything as far as end users are concerned, but it can break behavioral tests because Cypress by default fails a test if an uncaught error is thrown by the app, even if the app otherwise continues to function as expected by the test.  The error is no doubt thrown in a lot of other places for the same reason -- probably any place where there's an `<a href="#">` tag that triggers a JS action -- but this is the one I noticed.

The nifty scrolling that this code should have a behavioral test for it, but I opened this PR randomly and now I'm working on another issue primarily, so this might not see any movement for a few hours 😃

**Review by swirtSJW — APPROVED** [review]
Nice work Nate
